src/Parser/FormUrlencoded.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace React\Http\Parser;
+
+use Evenement\EventEmitterTrait;
+use React\Http\Request;
+
+class FormUrlencoded implements ParserInterface
+{
+    use EventEmitterTrait;
+
+    /**
+     * @var string
+     */
+    protected $buffer = '';
+
+    /**
+     * @var Request
+     */
+    protected $request;
+
+    /**
+     * @var bool|integer
+     */
+    protected $contentLength = false;
+
+    /**
+     * @param Request $request
+     */
+    public function __construct(Request $request)
+    {
+        $this->request = $request;
+
+        $headers = $this->request->getHeaders();
+        $headers = array_change_key_case($headers, CASE_LOWER);
+
+        $this->contentLength = $headers['content-length'];
+        $this->request->on('data', [$this, 'feed']);
+    }
+
+    /**
+     * @param string $data
+     */
+    public function feed($data)
+    {
+        $this->buffer .= $data;
+
+        if (
+            $this->contentLength !== false &&
+            strlen($this->buffer) >= $this->contentLength
+        ) {
```

Comment:
I feel like this is mixing concerns.. Shouldn't the whole body be delimited before passing it to the parsers?